ktor-client/ktor-client-ios/darwin/src/io/ktor/client/engine/ios/IosClientEngine.kt

```diff
@@ -123,6 +123,7 @@ internal class IosClientEngine(override val config: IosClientEngineConfig) : Htt
 
             config.requestConfig(nativeRequest)
             session.dataTaskWithRequest(nativeRequest).resume()
+            session.finishTasksAndInvalidate()
```

Comment:
The session is in progress after `resume()` call. Maybe we should call this in `invokeOnCompletion` of call context?

Reply:
Good idea; updated.